src/main/User/Services/GetUserService.java

```diff
@@ -0,0 +1,23 @@
+package User.Services;
+
+import Database.User.UserDao;
+import User.User;
+
+import java.util.Optional;
+
+public class GetUserService {
+
+  private UserDao userDao;
+
+  public GetUserService(UserDao userDao) {
+    this.userDao = userDao;
+  }
+
+  public User getUser(String username) {
+    Optional<User> result = userDao.get(username);
+    if (result.isEmpty()) {
+      return null;
```

Comment:
hmm i think this kinda defeats the point of Optional<>. Optional is a class that helps get rid of null pointers, but if we are returning nulls anyway, then its still a design flaw. In general, we should try to remove all instances where null pointers can happen so that we can always assume that an Object will be not null when it is passed from service to service. I would just remove GetUserService as an argument to many of the Service methods and pull straight from the DAO

Reply:
👍